src/api/nodeRpcCall.js

```diff
@@ -1924,6 +1924,63 @@ const fetchPendingIdentities = async () => {
   return processed.filter((entity) => entity.data.judgements.length === 0);
 };
 
+const getSignaturesForGivenId = async (allSignatures, id) => allSignatures.find(
+  ([contractId]) => contractId.args[0].eq(id),
+)[1].unwrapOrDefault();
+
+const getAllContracts = async () => {
+  const api = await getApi();
+
+  const [rawJudgesSignatures, rawPartiesSignatures] = await Promise.all([
+    api.query.contractsRegistry.judgesSignatures.entries(),
+    api.query.contractsRegistry.partiesSignatures.entries(),
+  ]);
+
+  const rawContracts = await api.query.contractsRegistry.contracts.entries();
+  return Promise.all(rawContracts.map(async ([id, contract]) => {
+    const contractId = id.args[0];
+    const [judgesSignatures, partiesSignatures] = await Promise.all([
+      getSignaturesForGivenId(rawJudgesSignatures, contractId),
+      getSignaturesForGivenId(rawPartiesSignatures, contractId),
+    ]);
+
+    const shouldBeData = contract.unwrapOr(null);
+    return {
+      contractId,
+      data: shouldBeData ? shouldBeData.data : null,
+      parties: shouldBeData ? shouldBeData.parties : null,
+      creator: shouldBeData ? shouldBeData.creator : null,
+      deposit: shouldBeData ? shouldBeData.deposit : null,
+      judgesSignatures,
+      partiesSignatures,
+    };
+  }));
+};
+
+const signContractAsParty = async (contractId, walletAddress) => {
+  const api = await getApi();
+  const extrinsic = api.tx.contractsRegistry.partySignContract(contractId);
+  return submitExtrinsic(extrinsic, walletAddress, api);
+};
+
+const signContractAsJudge = async (contractId, walletAddress) => {
+  const api = await getApi();
+  const extrinsic = api.tx.contractsRegistry.partySignContract(contractId);
+  return submitExtrinsic(extrinsic, walletAddress, api);
+};
+
+const removeContract = async (contractId, walletAddress) => {
+  const api = await getApi();
+  const extrinsic = api.tx.contractsRegistry.removeContract(contractId);
+  return submitExtrinsic(extrinsic, walletAddress, api);
+};
+
+const getAllJudges = async () => {
+  const api = await getApi();
+  const rawJudges = await api.query.contractsRegistry.judges.entries();
+  return rawJudges.map(([address, isJudge]) => isJudge.isTrue && address.args[0]).filter((address) => address);
```

Comment:
That's a weird order. Either swap filter and map:
```suggestion
  return rawJudges.filter(([, isJudge]) => isJudge.isTrue).map(([address, ]) => address.args[0]);
```
Or use reduce.